tests/unit/core/client/test_ConnectionManager.py

```diff
@@ -0,0 +1,235 @@
+            return err
+
+    def get_new_peers(self):
+        return self.peers_to_return
+
+class MocFunctionalQueryHandler(object):
+    implements(IQueryHandler)
+
+    def __init__(self, clock, is_good=True, is_delayed=False):
+        self.query_identifiers = ['moc_request']
+        self.is_good = is_good
+        self.is_delayed = is_delayed
+        self.clock = clock
+
+    def register_with_request_handler(self, request_handler, peer):
+        request_handler.register_query_handler(self, self.query_identifiers)
+
+    def handle_queries(self, queries):
+        if self.query_identifiers[0] in queries:
+            if self.is_delayed:
+                out = deferLater(self.clock, 10, lambda: {'moc_request':0})
+                self.clock.advance(10)
+                return out
+            if self.is_good:
+                return defer.succeed({'moc_request':0})
+            else:
+                return defer.succeed({'bad_request':0})
+        else:
+            return defer.succeed({})
+
+
+class MocQueryHandlerFactory(object):
+    implements(IQueryHandlerFactory)
+    # is is_good, the query handler works as expectd,
+    # is is_delayed, the query handler will delay its resposne
+    def __init__(self, clock, is_good=True, is_delayed=False):
+        self.is_good = is_good
+        self.is_delayed = is_delayed
+        self.clock = clock
+    def build_query_handler(self):
+        return MocFunctionalQueryHandler(self.clock, self.is_good, self.is_delayed)
+
+    def get_primary_query_identifier(self):
+        return 'moc_query'
+
+    def get_description(self):
+        return "This is a Moc Query"
+
+
+class MocServerProtocolFactory(ServerFactory):
+    protocol = ServerProtocol
+    def __init__(self, clock, is_good=True, is_delayed=False, has_moc_query_handler=True):
+        self.rate_limiter = RateLimiter()
+        query_handler_factory = MocQueryHandlerFactory(clock, is_good, is_delayed)
+        if has_moc_query_handler:
+            self.query_handler_factories = {
+                query_handler_factory.get_primary_query_identifier():query_handler_factory
+            }
+        else:
+            self.query_handler_factories = {}
+        self.peer_manager = PeerManager()
+
+
+class TestIntegrationConnectionManager(unittest.TestCase):
+    def setUp(self):
+
+        conf.initialize_settings()
+
+        self.TEST_PEER = Peer(LOCAL_HOST, PEER_PORT)
+        self.downloader = MocDownloader()
+        self.rate_limiter = RateLimiter()
+        self.primary_request_creator = MocRequestCreator([self.TEST_PEER])
+        self.connection_manager = ConnectionManager(self.downloader, self.rate_limiter,
+                                                    [self.primary_request_creator], [])
+
+        self.clock = task.Clock()
+        self.connection_manager.callLater = self.clock.callLater
+        self.connection_manager._start()
+        self.server_port = None
+
+    def tearDown(self):
```

Comment:
add in a `conf.settings = None` here so that the settings created in `setUp` don't affect other tests